packages/framework/src/Model/Transport/TransportRepository.php

```diff
@@ -170,12 +170,26 @@ public function getAllWithEagerLoadedDomainsAndTranslations(
 
         if ($totalWeight !== null) {
             $queryBuilder
-                ->andWhere('t.maxWeight IS NULL OR t.maxWeight >= :maxWeight')
+                ->join('t.prices', 'tp', Join::WITH, 'tp.domainId = :domainId')
+                ->andWhere('tp.maxWeight IS NULL OR tp.maxWeight >= :maxWeight')
                 ->setParameter('maxWeight', $totalWeight);
         }
 
         return $queryBuilder
             ->getQuery()
             ->getResult();
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport $transport
+     */
+    public function deleteAllPricesByTransport(Transport $transport): void
```

Comment:
@vitek-rostislav is it necessary? TransportPrice has `onDelete="CASCADE"`

Reply:
this is for updating the collection (the simple approach "delete all" -> "recreate everything"). I tried to implement the clever way of updating the collection with removing removed only, adding new only, and editing the edited ones, however, the code was too complicated and we agreed with  @grossmannmartin that it is not worth it